File: manual/data/python/tests/test_functools_partial.py
import unittest
from function_functools_partial import PartialFunctionExample

class TestPartialFunctionExample(unittest.TestCase):

    def test_apply_partial_function(self):
        calc = PartialFunctionExample()
        result = calc.apply_partial_function(5, 10)
        self.assertEqual(result, 15)

    def test_apply_partial_function_with_negative(self):
        calc = PartialFunctionExample()
        result = calc.apply_partial_function(-3, 7)
        self.assertEqual(result, 4)

    def test_apply_partial_function_zero(self):
        calc = PartialFunctionExample()
        result = calc.apply_partial_function(0, 5)
        self.assertEqual(result, 5)

    def test_non_numeric_input(self):
        calc = PartialFunctionExample()
        with self.assertRaises(TypeError):
            calc.apply_partial_function("five", 10)

    def test_edge_case(self):
        calc = PartialFunctionExample()
        result = calc.apply_partial_function(0, 0)
        self.assertEqual(result, 0)

if __name__ == '__main__':
    unittest.main()

File: data/java/tests/test_functools_partial.java
package com.example;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

public class test_functools_partial {

    @Test
    void test_apply_partial_function() {
        function_functools_partial calc = new function_functools_partial();
        int result = calc.apply_partial_function(5, 10);
        assertEquals(15, result);
    }

    @Test
    void test_apply_partial_function_with_negative() {
        function_functools_partial calc = new function_functools_partial();
        int result = calc.apply_partial_function(-3, 7);
        assertEquals(4, result);
    }

    @Test
    void test_apply_partial_function_zero() {
        function_functools_partial calc = new function_functools_partial();
        int result = calc.apply_partial_function(0, 5);
        assertEquals(5, result);
    }

    @Test
    void test_non_numeric_input() {
        function_functools_partial calc = new function_functools_partial();
        assertThrows(IllegalArgumentException.class, () -> {
            calc.apply_partial_function(Integer.valueOf((String)"five"), 10);
        });
    }

    @Test
    void test_edge_case() {
        function_functools_partial calc = new function_functools_partial();
        int result = calc.apply_partial_function(0, 0);
        assertEquals(0, result);
    }
}